### DISCUSSION TITLE
V2 Server

### DISCUSSION THREAD

**Discussion by o-alexandre-felipe** [doghappy/socket.io-client-csharp#226] [Q&A]
After I read [here](https://github.com/doghappy/socket.io-client-csharp#breaking-changes-in-300), that socket io V2 is no longer support.

Then I downgraded to version 2.2.0 in my implementation for the client to an existing server (EIO=3).

I see the pull request https://github.com/doghappy/socket.io-client-csharp/pull/218, and mentions to v2 support in https://github.com/doghappy/socket.io-client-csharp/commit/996b286baeab7890c8db636f4890109ce09655eb.

Is the latest version expected to work with server 2.4.1 ?

If so what are the settings we should set?

**Comment by doghappy** [comment]
The latest version v3.0.3 supports socket.io server v2/v3/v4. What you have to do is to know which version the server uses. If your server is using socket.io server v2.x, please explicitly set EIO to 3

```cs
            var socket = new SocketIO(uri, new SocketIOOptions
            {
                ...
                EIO = 3, // socket.io server v2
            });
```

---

In SocketIOClient v2 version, socket.io server v2/v3/v4 is supported. On the transport protocol, only websocket is supported.

In the SocketIOClient v3 version, support for http polling is added. For some reasons, socket.io server v2 is not supported. But I got a lot of feedback afterwards. The user was using socket.io server v2, so it returned to support socket.io server v2.
